src/components/Match/Chat/Chat.jsx

```diff
@@ -0,0 +1,294 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { isRadiant, formatSeconds } from 'utility';
+import strings from 'lang';
+import { IconRadiant, IconDire } from 'components/Icons';
+import AvVolumeUp from 'material-ui/svg-icons/av/volume-up';
+import Checkbox from 'material-ui/Checkbox';
+import Visibility from 'material-ui/svg-icons/action/visibility';
+import VisibilityOff from 'material-ui/svg-icons/action/visibility-off';
+import heroes from 'dotaconstants/build/heroes.json';
+import playerColors from 'dotaconstants/build/player_colors.json';
+import styles from './Chat.css';
+
+// All chat
+// https://github.com/dotabuff/d2vpkr/blob/8fdc29b84f3e7e2c130fc1b8c6ffe3b811e2d4a7/dota/scripts/chat_wheel.txt#L640
+const chatwheelAll = [75, 76, 108, 109, 110];
+
+class Chat extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.raw = this.props.data;
+
+    // detect spam
+    for (let i = 0; i < this.raw.length - 1; i += 1) {
+      const curr = this.raw[i];
+      const next = this.raw[i + 1];
+      if (curr.player_slot === next.player_slot) {
+        if ((next.time - curr.time) < 15) {
+          if (curr.key === next.key) {
+            next.spam = true;
+          }
+        }
+        // ex: 3334005345
+        if (curr.type === 'chat' && next.type === 'chat') {
+          // for some reason some strings have trailing space
+          curr.key = curr.key.trim();
+          next.key = next.key.trim();
+          // if first and last 2 chars matches, it's spam
+          if (curr.key.slice(0, 2) === next.key.slice(0, 2) && curr.key.slice(-2) === next.key.slice(-2)) {
+            next.spam = true;
+          }
+        }
+      }
+    }
+
+    this.state = {
+      radiant: true,
+      dire: true,
+      text: true,
+      phrases: true,
+      audio: true,
+      all: true,
+      allies: true,
+      spam: false,
+
+      playing: null,
+    };
+
+    this.filters = {
+      radiant: {
+        f: (arr = this.raw) => arr.filter(msg => isRadiant(msg.player_slot)),
+        type: 'faction',
+        disabled: () => !this.state.dire,
+      },
+      dire: {
+        f: (arr = this.raw) => arr.filter(msg => !isRadiant(msg.player_slot)),
+        type: 'faction',
+        disabled: () => !this.state.radiant,
+      },
+      text: {
+        f: (arr = this.raw) => arr.filter(msg => msg.type === 'chat'),
+        type: 'type',
+        disabled: () => this.state.phrases === false && this.state.audio === false,
+      },
+      phrases: {
```

Comment:
I've actually recently removed phrases from the data (leaving only audio chatwheel messages)